### ISSUE TITLE
Paginated Output Format's first page filepath is wrong

### ISSUE THREAD

**Issue by regisphilibert** [gohugoio/hugo#6797]
### What version of Hugo are you using (`hugo version`)?

<pre>
Hugo Static Site Generator v0.58.3/extended darwin/amd64 (or later)
</pre>

### Does this issue reproduce with the latest release?
Yes

----

When using paginated output formats on list templates, such as json. Hugo creates the following file:

```
/page/1/index.json/index.html
```

Other pages files are created as expected at:
```
/page/2/index.json
```

This might be what triggers #6472 

See this example repo (initially created for #6472) 👉 https://github.com/theNewDynamic/gohugo-6472

**Comment by bep** [comment]
OK, I have checked this, and:

* For HTML, we create alias pages for `/section/page/1` redirecting to `/section/`
* For JSON, a redirect obviously does not make sense
* But I'm not totally sure that duplicating so`/section/page/1/index.json == /section/index.json`  which would make a confusing setup.

My short term fix for this (I will keep this issue open) is to NOT creating alias for non-HTML paginators.

**Comment by regisphilibert** [comment]
> My short term fix for this (I will keep this issue open) is to NOT creating alias for non-HTML paginators.

I'm fine with that.

**Comment by ntrrg** [comment]
With v0.63.2 the `.Paginator` property for non-HTML formats is broken:

```
Building sites … ERROR 2020/02/02 11:57:57 render of "page" failed: execute of template failed: template: _default/list.json.json:4:37: executing "data" at <.Paginator.Pages>: error calling Pages: runtime error: invalid memory address or nil pointer dereference
```

Not sure if it is related to this

**Comment by ptgott** [comment]
Just checking to see if there's any more work to be done on this now that we've merged #6829.

**Comment by jmooring** [comment]
Resolved, presumably with #6829.

```bash
git clone --single-branch -b hugo-github-issue-6797 https://github.com/jmooring/hugo-testing hugo-github-issue-6797
cd hugo-github-issue-6797
hugo server
```

```text
public/posts/page/
├── 1/
│   └── index.html  <-- there isn't an alias for json, as expected/desired
├── 2/
│   ├── index.html
│   └── index.json
└── 3/
    ├── index.html
    └── index.json
```
